### PR TITLE
Fixed #33298 -- Added docs and tests for using Q objects with get_object_or_404()/get_list_or_404().

### PR DESCRIPTION
**Pull request by mgaligniana**
Ticket https://code.djangoproject.com/ticket/33298

Please let me know if they were the tests and documentation expected or any change needed!

Thank you!

### CODE REVIEW

File: docs/topics/http/shortcuts.txt [hunk 1]
```diff
@@ -166,6 +166,10 @@ Required arguments
     or a :class:`~django.db.models.query.QuerySet` instance from which to get
     the object.
 
+``*args``
+    Lookup parameters, which should be in the format accepted by ``get()`` and
+    ``filter()``.
```
Comment (felixxm) [inline docs/topics/http/shortcuts.txt:171]: I think it's enough to refer `Q` objects:
```suggestion
    :class:`Q objects <django.db.models.Q>`.
```

File: docs/topics/http/shortcuts.txt [hunk 2]
```diff
@@ -238,6 +242,10 @@ Required arguments
     :class:`~django.db.models.query.QuerySet` instance from which to get the
     list.
 
+``*args``
+    Lookup parameters, which should be in the format accepted by ``get()`` and
+    ``filter()``.
```
Comment (felixxm) [inline docs/topics/http/shortcuts.txt:247]:
```suggestion
    :class:`Q objects <django.db.models.Q>`.
```

File: tests/get_object_or_404/tests.py [hunk 3]
```diff
@@ -76,6 +77,26 @@ def test_get_object_or_404(self):
             [article]
         )
 
+        # get_object_or_404 can accept args
```
Comment (felixxm) [inline tests/get_object_or_404/tests.py:80]:
```suggestion
        # Q objects.
```

File: tests/get_object_or_404/tests.py [hunk 4]
```diff
@@ -76,6 +77,26 @@ def test_get_object_or_404(self):
...
+        self.assertEqual(
+            get_object_or_404(
+                Article,
+                Q(title__startswith="Run") | Q(title__startswith="Walk"),
+                authors__name__contains="Brave"
+            ),
+            article
+        )
+
+        # get_list_or_404 can accept args
```
Comment (felixxm) [inline tests/get_object_or_404/tests.py:90]: We can chop this comment.
```suggestion
```

File: tests/get_object_or_404/tests.py [hunk 5]
```diff
@@ -76,6 +77,26 @@ def test_get_object_or_404(self):
...
         )
 
+        # get_object_or_404 can accept args
+        self.assertEqual(
+            get_object_or_404(
+                Article,
+                Q(title__startswith="Run") | Q(title__startswith="Walk"),
+                authors__name__contains="Brave"
+            ),
+            article
```
Comment (felixxm) [inline tests/get_object_or_404/tests.py:87]: Use single quotes and add trailing commas,
```suggestion
            get_object_or_404(
                Article,
                Q(title__startswith='Run') | Q(title__startswith='Walk'),
                authors__name__contains='Brave',
            ),
            article,
```

File: tests/get_object_or_404/tests.py [hunk 6]
```diff
@@ -76,6 +77,26 @@ def test_get_object_or_404(self):
...
+        )
+
+        # get_list_or_404 can accept args
+        self.assertEqual(
+            get_list_or_404(
+                Article,
+                Q(title__startswith="Run") | Q(title__startswith="Walk"),
+                authors__name__contains="Patsy"
+            ),
+            [article]
```
Comment (felixxm) [inline tests/get_object_or_404/tests.py:97]:
```suggestion
            get_list_or_404(
                Article,
                Q(title__startswith='Run') | Q(title__startswith='Walk'),
                authors__name='Patsy',
            ),
            [article],
```

**Review by felixxm — COMMENTED** [review]
@mgaligniana Thanks for this patch :+1: I left small comments.

**Comment by jaap3** [comment]
Also, should an example using `Q` objects be added?

**Comment by felixxm** [comment]
> Also, should an example using `Q` objects be added?

`Q` objects references have a link to the ["Complex lookups with Q objects"](https://docs.djangoproject.com/en/stable/topics/db/queries/#complex-lookups-with-q-objects) with many examples, so I think there is no need to add more examples here.

**Review by felixxm — COMMENTED** [review]
@mgaligniana Thanks :+1: Welcome aboard :boat:

**Pull request merged**